### PR TITLE
Adding the Transaction Service topics

### PR DESCRIPTION
**Pull request by dafo**

### CODE REVIEW

File: en/components/transaction-how-to.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+* [Hierarchical Grid Row Editing](hierarchicalgrid/row_editing.md)
+
+### General information on igxTransactionService and igxHierarchicalTransactionService
+
+`igxTransactionService` and `igxHierarchicalTransactionService` are injectable middlewares a component can use to accumulate changes without affecting the underlying data. The provider exposes API to access, manipulate changes (undo and redo) and discard or commit all to the data.
+
+`igxTransactionService` and `igxHierarchicalTransactionService` work with both cell editing and row editing. The transaction for the cell edit is added when the cell exits edit mode, while row transaction is created, when the row exits edit mode. But in both cases the state of the grid edits consist of all updated, added and deleted rows and their last states. Those can later be inspected, manipulated and submitted at once. Changes are collected for individual cells or rows, depending on editing mode, and accumulated per data row/record.
+
+### Using igxTransactionService
+
```
Comment (dafo) [inline en/components/transaction-how-to.md]: Describe that it implements the BaseTransaction

File: en/components/transaction.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+_description: 
+_keywords: Ignite UI for Angular, transaction
+---
+
+
+## Transaction Service
+
+The Transaction Service is an injectable middleware (through Angular's DI) that a component may use to accumulate changes without immediately affecting the underlying data.
+    <img src="https://cdn-images-1.medium.com/max/800/1*O-6DidcFW_XCSqgKRfXf_Q.png"
+        style="display:flex;max-height:400px;margin:auto auto 20px auto;" />
```
Comment (Lipata) [inline en/components/transaction.md]: Note that `transform data through pipe` in the image is not a mandatory operation.

File: en/components/transaction.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+The Transaction Service allows adding a transaction, committing or clearing all changes. As it keeps a detailed log, it can also execute undo and redo operations.
```
Comment (Lipata) [inline en/components/transaction.md]: As it keeps a detailed log `(state)` / `called state`. At some point, we should introduce the `state` terminology, here could be one of the places.

File: en/components/transaction.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+Every time you execute an operation (transaction), it is added to the transaction log and undo stack. All the changes in the transaction log are then accumulated per record. From that point, the service maintains an aggregated state that consists only of add/update/delete operations for unique records.
```
Comment (Lipata) [inline en/components/transaction.md]: we can bold **transaction** and **state** at least in this topic, eventually in all others?

File: en/components/transaction-example.md [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+Our class constructor should look like this:
+
+`constructor(private _transactions: IgxTransactionService<Transaction, State>) { ... }`
+
+When we click the `add` button, we add a new transaction to the Transaction log, providing the `id`, the `TransactionType` and the `newValue` properties: 
+
+`this._transactions.add({ id: 2, type: TransactionType.ADD, newValue: person });`
+
+At this moment, our transaction is added to the transaction log, we see the new item is added to the list that we are manipulating but the change is still not committed. We may do additional changes to the list and commit them at once when we are ready.
```
Comment (wnvko) [inline en/components/transaction-example.md]: I think we should boldly point that data source is still unchanged like this

... is still not committed. The data source is not changed yet. We may...

File: en/components/transaction-how-to.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+The `id` of the Transaction should be unique and defines the record that this transaction is affecting. The `type` may be any of the three transaction types: `ADD`, `DELETE` and `UPDATE`, depending what operation you execute. The `newValue` contains the value of the new record in case you are adding an `ADD` transaction. If you are editing an existing record, the `newValue` would contain the changes only. If you are deleting a record, the `newValue` will contain the value of the deleted record. 
+
+You may see an example of how adding each type of transaction looks like in the [How to use the Transaction service](transaction-example.md) topic.
+
+Every time you add a Transaction, it is added to the transaction log and undo stack. All the changes in the transaction log are then accumulated per record. From that point, the service maintains an aggregated `State`. The `State`consists of unique records and every record may be of one of the supported transaction types, mentioned above.
+
```
Comment (wnvko) [inline en/components/transaction-how-to.md]: We should include something about pending transactions here. Please add this or something better:

While adding transactions you may turn on pending transactions by calling `startPending`. All subsequent transactions will be accumulated in single transaction until you call `endPending`. If you pass `true` to `endPending` all accumulated  transactions will be added as a single transaction in transaction log and in undo stack.

File: en/components/transaction-how-to.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+The `igxBaseTransactionService` has no undo stack so it does not provide undo/redo functionality.
+
+A detailed example of how you may use `igxBaseTransaction` to enable Row Editing is provided in the following topics:
+* [Grid Row Editing](grid/row_editing.md)
+* [Tree Grid Row Editing](treegrid/row_editing.md)
+* [Hierarchical Grid Row Editing](hierarchicalgrid/row_editing.md)
+
+### General information on igxTransactionService and igxHierarchicalTransactionService
+
+`igxTransactionService` and `igxHierarchicalTransactionService` are injectable middlewares, that implement the `TransactionService` interface. A component may use those to accumulate changes without affecting the underlying data. The provider exposes API to access, manipulate changes (undo and redo) and discard or commit all to the data.
```
Comment (wnvko) [inline en/components/transaction-how-to.md]: You can discard or commit and single change. Add this:

`... and discard or commit all or single change to the data.`

File: en/components/transaction-example.md [hunk 8]
```diff
@@ -0,0 +1,58 @@
+---
+title: How to use the Transaction service
+_description: 
+_keywords: Ignite UI for Angular, transaction
+---
+
+## How to use the Transaction service
+
+You may get advantage of the Transaction Service when using any control that needs to preserve state and to commit many transactions at once.
```
[the remark this answers is not in the document]
Comment (dafo) [reply en/components/transaction-example.md]: Also, change 'control' to 'component'

File: en/components/transaction-how-to.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+* [Grid Row Editing](grid/row_editing.md)
+* [Tree Grid Row Editing](treegrid/row_editing.md)
+* [Hierarchical Grid Row Editing](hierarchicalgrid/row_editing.md)
+
+### General information on igxTransactionService and igxHierarchicalTransactionService
+
+`igxTransactionService` and `igxHierarchicalTransactionService` are injectable middlewares, that implement the `TransactionService` interface. A component may use those to accumulate changes without affecting the underlying data. The provider exposes API to access, manipulate changes (undo and redo) and discard or commit all changes or a single change to the data.
+
+`igxTransactionService` and `igxHierarchicalTransactionService` work with both cell editing and row editing. The transaction for the cell edit is added when the cell exits edit mode, while row transaction is created, when the row exits edit mode. But in both cases the state of the grid edits consist of all updated, added and deleted rows and their last states. Those can later be inspected, manipulated and submitted at once. Changes are collected for individual cells or rows, depending on editing mode, and accumulated per data row/record.
```
Comment (wnvko) [inline en/components/transaction-how-to.md]: `The transaction for the cell edit is added when the cell exits edit mode, while row transaction is created, when the row exits edit mode.`

I would separate this into two sentences and add some additional information about row editing. Something like this:

`The transaction for the cell edit is added when the cell exits edit mode. When row editing starts the grid set transaction service in pending state by calling startPendting. Each edited cell is added to the pending transaction log and is not added to the main transaction log. When the row exits edit mode all the changes are added to the main transaction log and to the undo log as a single transaction.`

Comment (wnvko) [inline en/components/transaction-how-to.md]: `Those can later be inspected, manipulated and submitted at once`

you can also submit and per id transaction, please add it here. Something like this:

`Those can later be inspected, manipulated and submitted at once or per id`

File: en/components/transaction-how-to.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+The `igxTransactionService` provides an undo stack so you may get advantage of the undo/redo functionality. The Undo stack is actually an array that contains array of transactions. When using the `igxTransactionService`, you may check the `canUndo` accessor in order to understand if there are any transactions in the Undo stack. If it does - you may use the `undo` method to remove the last transaction and `redo` to apply the last undone transaction.
+
+You may find a detailed example of how igxGrid with Batch Editing is implemented in the following topic:
+* [Grid Batch Editing](grid/batch_editing.md)
+
+### Using igxHierarchicalTransactionService
+
+`igxHierarchicalTransactionService` extends `IgxTransactionService`.
+
+The `igxHierarchicalTransactionService` is designed to handle the relations between parents and children in `igxHierarchicalGrid` and `igxTreeGrid` so that a new record will be added to the place you expect when adding an `ADD` transaction. And when you delete a parent record, its' children will be promoted to the higher level of hierarchy.
```
Comment (wnvko) [inline en/components/transaction-how-to.md]: `, its' children will be promoted to the higher level of hierarchy.`

the children could also be deleted. Change this to something like this:

`, its' children will be promoted to the higher level of hierarchy, or will be deleted with their parent.`

File: en/components/transaction-how-to.md [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+_keywords: Ignite UI for Angular, transaction
+---
+
+## igxBaseTransaction, igxTransaction and igxHierarchical transaction explained
+
+### Transaction, State, Transaction Log
+
+The Transaction is the main building block of the Transaction service. The Transaction is actually every operation that you execute on the data. The `Transaction` interface defines three properties: `id`, `newValue` and `type`.
+
+The `id` of the Transaction should be unique and defines the record that this transaction is affecting. The `type` may be any of the three transaction types: `ADD`, `DELETE` and `UPDATE`, depending what operation you execute. The `newValue` contains the value of the new record in case you are adding an `ADD` transaction. If you are editing an existing record, the `newValue` would contain the changes only. If you are deleting a record, the `newValue` will contain the value of the deleted record.
```
Comment (wnvko) [inline en/components/transaction-how-to.md]: The `id` should be unique per data record. You may have several transactions of `Update` type with same `id`.

**Pull request merged**